kernel.go

```diff
@@ -306,25 +335,15 @@ func handleExecuteRequest(ir *classic.Interp, receipt msgReceipt) error {
 	wErr.Close()
 	stdErr := <-outStderr
 
+	// TODO write stdout and stderr to streams rather than publishing as results
+
 	if len(val) > 0 {
 		content["status"] = "ok"
-		content["payload"] = make([]map[string]interface{}, 0)
-		content["user_variables"] = make(map[string]string)
 		content["user_expressions"] = make(map[string]string)
-		if !silent {
-			var outContent OutputMsg
-
-			out, err := NewMsg("execute_result", receipt.Msg)
-			if err != nil {
-				return err
-			}
 
-			outContent.Execcount = ExecCounter
-			outContent.Data = make(map[string]string)
-			outContent.Data["text/plain"] = val
-			outContent.Metadata = make(map[string]interface{})
-			out.Content = outContent
-			receipt.SendResponse(receipt.Sockets.IOPubSocket, out)
+		if !silent {
+			// Publish the result of the execution
+			receipt.PublishExecutionResult(ExecCounter, val)
```

Comment:
error handling for Publish?

Reply:
For handling a _publish_ error should the entire request fail or should the handle log something but continue? The publish messages are usually stateless such as the `stream` messages and if it fails there is still a possibility to recover.

Reply:
Yeah, that makes sense.  We can `log.Println` these errors or similar.  At some point (maybe post v1), we should consider a more complete strategy for logging, verbosity, etc.